### ISSUE TITLE
EmitterMixin: try to get rid of `ck-expando` attribute.

### ISSUE THREAD

**Issue by scofalik** [ckeditor/ckeditor5-utils#120]
We are using `ck-expando` attribute on `HTMLElement`s to write unique emitter id in it, so then it is easy to recognize which node is connected with which `ProxyEmitter`.

It would be cleaner if we do not use this solution. Instead we could use some `WeakMap` that would connect `HTMLElement`s with `ProxyEmitter`s without directly using `_emitterId`. This would make DOM cleaner (obviously) and code cleaner (i.e. there would be no need to use `_emitterId` outside of `EmitterMixin` - now it is used by `ProxyEmitter`).

This is not something super urgent, though.

**Comment by Reinmar** [comment]
Another point is that it should not be `ck` but `cke`.

**Comment by Reinmar** [comment]
We realised in https://github.com/ckeditor/ckeditor5-utils/pull/118#pullrequestreview-16124522 that using a `WeakMap` will also allow to improve the `emittermixin` module.

Right now, the `ProxyEmitter` sets its own `_emitterId` to `getNodeUID( node )` in order to "inherit" this node's events.

This means that we need a setter for `_emitterId`, which is ugly. It'd be better if we only has a getter which, if the symbol property is not yet set, sets it and returns that value. We'll be able to do that if creation of the `proxyEmitter`s instances is done differently:

```js
DomEmitterMixin = {
  listenTo( emitter, ... ) {
    if ( isDomNode( emitter ) ) {
          emitter = getNodeEmitter( emitter );
    }

    super.listenTo( emitter, ... );
  }
};

const nodeEmitters = new WeakMap();

function getNodeEmitter( node ) {
  if ( !nodeEmitters.has( node ) ) {
    nodeEmitters.set( node, Object.create( EmitterMixin ) );
  }

  return nodeEmitters.get( node );
}
```

**Comment by Reinmar** [comment]
**tl;dr**: When fixing this, remove the `_setEmitterId()` from the `utils/emittermixin` module and make the `_getEmitterId()` function set the symbol propety on first use.
